src/kesslerav/protocol2k/io.py

```diff
@@ -251,33 +258,33 @@ def process(self, instructions: list[Instruction] | Instruction) -> list[Instruc
     # Results is a list of lists, so flatten before returning
     flat_results = list(itertools.chain.from_iterable(results))
     return flat_results
-  
+
   def _create_connection(self) -> socket.socket:
     return socket.create_connection(
       (self._endpoint.host, self._endpoint.port),
       self._endpoint.timeout_sec
     )
-  
+
   def _execute_instruction(
       self,
       instruction: Instruction,
       conn: socket.socket
     ) -> list[Instruction]:
-    req_bytes = Codec.encode(instruction)
+    req_bytes= Codec.encode(instruction)
     conn.send(req_bytes)
-
     # Device can return multiple instructions when its physical controls are
     # used. To capture them all (to reconstruct device state) we read using a
     # buffer that can hold multiple instructions and then return them all in
     # chronological event order.
-    result: list[Instruction] = []
+    result: list[Instruction]= []
     try:
       while len(result) < 1:
-        data = conn.recv(TcpDevice.BUFFER_SIZE_BYTES)
-        responses = struct.iter_unpack(Instruction.FORMAT, data)
+        time.sleep(0.5) # Slow down to prevent 'port already in use' errors when interacting with serial over TCP TODO configurable, default off
```

Comment:
@krohrbaugh this is that forced slowdown - without it my matrix was dropping quite a few commands.

I think I need to figure out how to get ser2net to add a buffer/slowdown on that side instead.